### PR TITLE
implement some hooks

### PR DESCRIPTION
**Pull request by ilyaagarkov**
https://favro.com/organization/aee06c3fe6503c804f507dd9/22c630cc02b09f2d6f98b8b3?card=Goo-25731

### CODE REVIEW

File: src/adaptive/index.es [hunk 1]
```diff
@@ -1,35 +1,20 @@
...
-    componentDidMount() {
-      this.detectDevice();
-      this.stopListeningToResize = eventproxy('resize', this.detectDevice);
-    }
-
-    componentWillUnmount() {
-      this.stopListeningToResize();
-    }
+export const useIsMobile = () => {
+  const [isMobile, setIsMobile] = useState(false);
```
Comment (max-degterev) [inline src/adaptive/index.es]: not sure if `null` is needed, perhaps default to `null` to signify unset state?

File: src/adaptive/index.es [hunk 2]
```diff
@@ -1,35 +1,20 @@
...
+  return isMobile;
+};
 
-    render() {
-      const { isMobile } = this.state;
-      const { forwardedRef, ...props } = this.props;
-      return <Component {...props} ref={forwardedRef} mobile={isMobile} />;
-    }
+const makeAdaptive = (Component, options = {}) => {
+  function AdaptiveComponent({ forwardedRef, ...props }) {
```
Comment (max-degterev) [inline src/adaptive/index.es]: const

File: src/history/index.es [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
 import { getDisplayName } from '../utils';
 
 
 const withHistory = (Component) => {
   const displayName = getDisplayName('withHistory', Component);
 
-  const WrappedComponent = ({ forwardedRef, ...props }) => (
-    <Component {...props} ref={forwardedRef} />
-  );
+  const WrappedComponent = ({ forwardedRef, ...props }, ref) => {
```
Comment (max-degterev) [inline src/history/index.es]: forwardedRef is no longer used

File: src/i18n/index.es [hunk 4]
```diff
@@ -1,15 +1,18 @@
-import React, { PureComponent } from 'react';
+import React, { PureComponent, useContext } from 'react';
 import Polyglot from 'node-polyglot';
 import LocaleContext from './context';
 import { getDisplayName, getForwardedComponent } from '../utils';
 
 const { Provider, Consumer } = LocaleContext;
 
+export function useInternational() {
```
Comment (max-degterev) [inline src/i18n/index.es]: const

File: package.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
   "homepage": "https://nebenan.de/",
   "repository": "good-hood-gmbh/nebenan-react-hocs",
   "bugs": "https://github.com/good-hood-gmbh/nebenan-react-hocs/issues",
-  "version": "5.2.2",
+  "version": "5.3.0",
```
Comment (max-degterev) [inline package.json:10]: let's publish -beta.1 just to be safe

File: src/history/index.es [hunk 6]
```diff
@@ -1,23 +1,26 @@
...
+    const history = useHistory();
+    const location = useLocation();
 
-  const WithRouter = withRouter(WrappedComponent);
-  const WithRef = (props, ref) => <WithRouter {...props} forwardedRef={ref} />;
+    return (
+      <Component history={history} location={location} ref={ref} {...props} />
```
Comment (max-degterev) [inline src/history/index.es]: match is still needed in some components, for example Dropdown/Form

**Review by max-degterev — CHANGES_REQUESTED** [review]

File: src/adaptive/index.es (hunk 1 above)
Comment (ilyaagarkov) [reply src/adaptive/index.es]: it looks like we needed `null` only for ssr

Comment (max-degterev) [reply src/adaptive/index.es]: if it's still in use please default to `null`

File: src/history/index.es [hunk 7]
```diff
@@ -1,23 +1,34 @@
...
+    const history = useHistory();
+    const location = useLocation();
+    const params = useParams();
+    const match = { params };
 
-  const WithRouter = withRouter(WrappedComponent);
-  const WithRef = (props, ref) => <WithRouter {...props} forwardedRef={ref} />;
+    return (
+      <Component
+        history={history}
```
Comment (max-degterev) [inline src/history/index.es:19]: {...props} {...{ history, location, match, ref }}

File: src/history/index.es [hunk 8]
```diff
@@ -26,5 +37,4 @@ export const historyPropTypes = {
   history: PropTypes.object.isRequired,
   location: PropTypes.object.isRequired,
   match: PropTypes.object.isRequired,
-  staticContext: PropTypes.object,
```
Comment (max-degterev) [inline src/history/index.es]: put back

File: src/adaptive/index.es [hunk 9]
```diff
@@ -1,36 +1,21 @@
...
-      this.detectDevice();
-      this.stopListeningToResize = eventproxy('resize', this.detectDevice);
-    }
-
-    componentWillUnmount() {
-      this.stopListeningToResize();
-    }
+export const useIsMobile = () => {
+  const [isMobile, setIsMobile] = useState(null);
+  useEffect(() => eventproxy('resize', () => setIsMobile(!getMedia(window, media.mediaM))), []);
```
Comment (atomiomi) [inline src/adaptive/index.es:9]: unsubscribe

File: src/adaptive/index.es [hunk 10]
```diff
@@ -1,36 +1,21 @@
...
-    componentDidMount() {
-      this.detectDevice();
-      this.stopListeningToResize = eventproxy('resize', this.detectDevice);
-    }
-
-    componentWillUnmount() {
-      this.stopListeningToResize();
-    }
+export const useIsMobile = () => {
+  const [isMobile, setIsMobile] = useState(null);
```
Comment (atomiomi) [inline src/adaptive/index.es]: default false/true

File: src/adaptive/index.es [hunk 11]
```diff
@@ -1,36 +1,21 @@
...
+  return isMobile;
+};
 
-    render() {
-      const { isMobile } = this.state;
-      const { forwardedRef, ...props } = this.props;
-      return <Component {...props} ref={forwardedRef} mobile={isMobile} />;
-    }
-  }
+const makeAdaptive = (Component, options = {}) => {
```
Comment (atomiomi) [inline src/adaptive/index.es]: withIsMobile?

File: src/i18n/index.es [hunk 12]
```diff
@@ -1,15 +1,16 @@
-import React, { PureComponent } from 'react';
+import React, { PureComponent, useContext } from 'react';
 import Polyglot from 'node-polyglot';
 import LocaleContext from './context';
 import { getDisplayName, getForwardedComponent } from '../utils';
 
 const { Provider, Consumer } = LocaleContext;
 
+export const useInternational = () => useContext(LocaleContext);
```
Comment (atomiomi) [inline src/i18n/index.es:8]: useLocale?

**Review by atomiomi — CHANGES_REQUESTED** [review]

File: src/adaptive/index.es (hunk 9 above)
Comment (ilyaagarkov) [reply src/adaptive/index.es:9]: callback passed to useEffect can return a function will be called on unmounted

**Review by max-degterev — APPROVED** [review]

**Pull request merged**